plugins/modules/docker_image_build.py

```diff
@@ -248,35 +323,86 @@ def add_args(self, args):
         if self.target:
             args.extend(['--target', self.target])
         if self.platform:
-            args.extend(['--platform', self.platform])
+            for platform in self.platform:
+                args.extend(['--platform', platform])
         if self.shm_size:
             args.extend(['--shm-size', str(self.shm_size)])
         if self.labels:
             self.add_list_arg(args, '--label', dict_to_list(self.labels))
+        if self.secret:
+            for secret in self.secret:
+                secret_str = 'id={id},src={src}'.format(id=secret["id"], src=secret["src"])
+                args.extend(['--secret', secret_str])
+        if self.load:
+            args.append('--load')
+        if self.push:
+            args.append('--push')
+
+    def parse_platforms(self, platform_param):
+        platforms = []
+        if platform_param:
+            if isinstance(platform_param, list):
+                platforms = platform_param
+            elif isinstance(platform_param, str):
+                try:
+                    parsed = json.loads(platform_param)
+                    if isinstance(parsed, list):
+                        platforms = parsed
+                    else:
+                        platforms.append(parsed)
+                except ValueError:
+                    platforms.append(platform_param)
+            else:
+                self.fail("Invalid platform format. Expected string, JSON list, or YAML list.")
+        return platforms
+
+    def handle_secrets(self):
+        secrets = self.secret or []
+        temp_files = []
+        for secret in secrets:
+            if 'value' in secret:
```

Comment:
This is **always** true. `secret['value']` is either a string or `None`, depending on whether the user provided a value here or not.